### ISSUE TITLE
Support Ubuntu 22.04 properly (Error: `libcrypto.so.1.1 missing`)

### ISSUE THREAD

**Issue by lyj711** [nodkz/mongodb-memory-server#732]
## Versions

- NodeJS: 16.13.0
- mongodb-memory-server-*: 8.11.0 
- mongodb(the binary version): 6.0.0 
- mongodb(the js package): 4.13.0
- system: Ubuntu

package: mongo-memory-server

## What is the Problem?
When running 'pnpm test', following error occured:

Instance failed to start because a library is missing or cannot be opened: "libcrypto.so.1.1"

## Code Example

```ts
import { Test, TestingModule } from '@nestjs/testing';
import { EventController } from './event.controller';
import { EventService } from './event.service';
import {MongoMemoryReplSet} from 'mongodb-memory-server';
import {Event} from '@prisma/client';
import { PrismaService } from '../prisma.service';
import { BadRequestException } from '@nestjs/common';
import { BlockchainService } from '../blockchain/blockchain.service';


describe('EventController', () => {
  let controller: EventController;
  let prisma: PrismaService;
  let mongodb: MongoMemoryReplSet;
  let eventModel: Event;

  beforeAll(async () => {
    mongodb = await MongoMemoryReplSet.create();
  });

  beforeEach(async () => {
    process.env.DATABASE_URL = mongodb.getUri('test');

    const module: TestingModule = await Test.createTestingModule({
      controllers: [EventController],
      providers: [EventService,PrismaService,BlockchainService],
    }).compile();

    controller = module.get<EventController>(EventController);
    eventModel = module.get<Event>(Event);
  });

  afterEach(async () => {
    await prisma.event.deleteMany({});

  })

  afterAll(async () => {
    await mongodb.stop();
  });



  it('Should throw an error if event id is not valid',async  () => {
    await expect(() => controller.getEvent('63bbd31c7a644b96fc34dc28')).toThrow(
      BadRequestException);
  });
});

```

## Debug Output

<details>
<summary>Debug Output</summary>

```txt
EventController › Should throw an error if event id is not valid

    Instance failed to start because a library is missing or cannot be opened: "libcrypto.so.1.1"
```

</details>

## Do you know *why* it happenes?

*no*

**Comment by hasezoey** [comment]
known issue, nothing MMS can do about - there are just no Ubuntu 2204 binaries yet, so MMS falls back to 2004 binaries, but those require OpenSSL 1.1, and Ubuntu 2204 only ships 3.x, so your options are:
- downgrade ubuntu version
- manually install openssl 1.1
- wait for mongodb to drop a binary for 2204
- use the mongodb packages from APT and set a system binary environment variable as a workaround *1

see https://jira.mongodb.org/browse/SERVER-62300
TL;DR: they are *planning* to add a ubuntu 2204 binary for 6.0.4

*1 somehow in ubuntu APT archives, there are mongodb builds for ubuntu 2204, but no "official" mongodb builds ([Official builds can be found here](https://www.mongodb.com/download-center/community/releases/archive))

**Comment by MichaelJCole** [comment]
See also: https://github.com/shelfio/jest-mongodb/issues/408

I have mongodb running all over the place on 22.04 with docker.  

Would it be possible to substitute the binary download for a docker command?

**Comment by hasezoey** [comment]
> Would it be possible to substitute the binary download for a docker command?

MMS is currently not build to use anything other than the direct binary, you currently have 2 options:
- automatic download, assuming the binary exists there
- use a manual binary and let MMS use that (namely [`SYSTEM_BINARY`](https://nodkz.github.io/mongodb-memory-server/docs/api/config-options#system_binary) config option)

i dont think MMS will add the ability to use anything other than the direct binary, though you could also just run MMS *inside* a container and let it use a binary there

**Comment by hasezoey** [comment]
update, there is now mongodb 6.0.4, which has a build for ubuntu 2204 (though that version is seemingly not documented yet)

[build 6.0.4 ubuntu2204](https://fastdl.mongodb.org/linux/mongodb-linux-x86_64-ubuntu2204-6.0.4.tgz)

**Comment by hasezoey** [comment]
Note: for now it still required to set a custom mongodb version with at least `6.0.4` (currently default for MMS `8.x` is still mongodb `5.0.x`, see [config option `VERSION`](https://nodkz.github.io/mongodb-memory-server/docs/api/config-options#version)

see [When a Version gets upgraded](https://nodkz.github.io/mongodb-memory-server/docs/guides/mongodb-server-versions#when-a-version-gets-upgraded)

**Comment by forivall** [comment]
if someone needs 6.0.3 for any weird reason, i found that the .deb download contains a version which should work https://github.com/nodkz/mongodb-memory-server/issues/686#issuecomment-1370468664

**Comment by Ikusa0** [comment]
I'm still having this error even with the latest versions of MongoDB.

**Versions:**

NodeJS: 19.6.0
mongodb-memory-server-*: 8.11.5
mongodb (binary): 6.0.5
mongodb (package): 5.5.0
system: POP_OS (Ubuntu jammy (22.04) like distro)

**jest-mongodb-config.js:**

    module.exports = {
      mongodbMemoryServerOptions: {
        binary: {
          version: '6.0.5',
          skipMD5: true
        },
        instance: {
          dbName: 'jest'
        },
        autoStart: false
      }
    }

**Comment by sagrawal31** [comment]
Still facing the problem in GitLab `node:lts` on GitLab CI (after setting `export MONGOMS_VERSION=6.0.4`)-

```
Running with gitlab-runner 16.1.0~beta.5.gf131a6a2 (f131a6a2)
  on green-5.shared.runners-manager.gitlab.com/default xS6Vzpvo, system ID: s_d878f787d8ea
  feature flags: FF_USE_IMPROVED_URL_MASKING:true
Preparing the "docker+machine" executor
Using Docker executor with image node:lts ...
```

**Comment by sagrawal31** [comment]
Thanks, @hasezoey. I changed to `node:lts-bullseye` and it worked.

**Comment by olof-nord** [comment]
This error is occurring for me too - trying to upgrade from Node.js v18.18.2 to v20.

    NodeJS: 20.9.0
    mongodb-memory-server-*: 9.1.1
    mongodb(the binary version): 6.0.6
    mongodb(the js package): 5.9.2
    system: Ubuntu

does anyone have any ideas how to resolve this?

**Comment by hasezoey** [comment]
> This error is occurring for me too - trying to upgrade from Node.js v18.18.2 to v20.

this issue has nothing to do with a nodejs version upgrade, if you have such issues, please open a new issue

**Comment by olof-nord** [comment]
> > This error is occurring for me too - trying to upgrade from Node.js v18.18.2 to v20.
> 
> this issue has nothing to do with a nodejs version upgrade, if you have such issues, please open a new issue

Many thanks for the quick reply. So it turns out the problem was somewhere else (CI) - the search results led me here though.
